src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,21 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Machine> {
```

Comment:
Parameterize `MachineProducer` in such a way that any implementation NOT using `Machine` or its subtype could NOT be created. In this case, you should use `Truck` instead of `Machine` as the generic type parameter for `MachineProducer`.